launcher/ui/widgets/ModFilterWidget.cpp

```diff
@@ -358,18 +391,120 @@ void ModFilterWidget::onOpenSourceFilterChanged()
 void ModFilterWidget::onReleaseFilterChanged()
 {
     std::list<ModPlatform::IndexedVersionType> releases;
-    if (ui->releaseCb->isChecked())
+    if (m_ui->releaseCb->isChecked())
         releases.push_back(ModPlatform::IndexedVersionType(ModPlatform::IndexedVersionType::VersionType::Release));
-    if (ui->betaCb->isChecked())
+    if (m_ui->betaCb->isChecked())
         releases.push_back(ModPlatform::IndexedVersionType(ModPlatform::IndexedVersionType::VersionType::Beta));
-    if (ui->alphaCb->isChecked())
+    if (m_ui->alphaCb->isChecked())
         releases.push_back(ModPlatform::IndexedVersionType(ModPlatform::IndexedVersionType::VersionType::Alpha));
-    if (ui->unknownCb->isChecked())
+    if (m_ui->unknownCb->isChecked())
         releases.push_back(ModPlatform::IndexedVersionType(ModPlatform::IndexedVersionType::VersionType::Unknown));
     m_filter_changed = releases != m_filter->releases;
     m_filter->releases = releases;
     if (m_filter_changed)
         emit filterChanged();
 }
 
+void ModFilterWidget::saveSetting(QString id, QVariant value)
+{
+    auto const setting_name = getSettingId(id);
+    auto setting = m_instance->settings()->getOrRegisterSetting(setting_name);
+
+    setting->set(value);
+}
+
+QVariant ModFilterWidget::getSetting(QString id)
+{
+    auto const setting_name = getSettingId(id);
+    auto setting = m_instance->settings()->getOrRegisterSetting(setting_name);
+
+    return setting->get();
+}
+
+void ModFilterWidget::saveFilters()
+{
+    if (!m_instance) {
+        return;
+    }
+    QStringList versionStrings;
+    for (const auto& v : m_filter->versions)
+        versionStrings.append(v.toString());
+
+    QStringList releaseStrings;
+    for (const auto& r : m_filter->releases)
+        releaseStrings.append(r.toString());
+
+    QStringList loaders;
+    for (auto loader : ModPlatform::modLoaderTypesToList(m_filter->loaders)) {
+        loaders.push_back(getModLoaderAsString(loader));
+    }
+    if (!versionStrings.isEmpty() || !loaders.isEmpty() || !m_filter->categoryIds.isEmpty() || !releaseStrings.isEmpty()) {
+        saveSetting("versions", versionStrings);
+        saveSetting("releases", releaseStrings);
+        saveSetting("loaders", loaders);
+        saveSetting("side", m_filter->side);
+        saveSetting("hideInstalled", m_filter->hideInstalled);
+        saveSetting("categoryIds", m_filter->categoryIds);
+        saveSetting("openSource", m_filter->openSource);
+        saveSetting("filtersSet", true);
+    }
+}
+
+void ModFilterWidget::loadFilters()
+{
+    if (!m_instance || !getSetting("filtersSaved").toBool() || !getSetting("filtersSet").toBool()) {
```

Comment:
I  don't think there's any reason to dynamically register filtersSaved and filtersSet